aws/table_aws_networkfirewall_firewall.go

```diff
@@ -0,0 +1,290 @@
+				Type:        proto.ColumnType_BOOL,
+				Hydrate:     getNetworkFirewallFirewall,
+				Transform:   transform.FromField("Firewall.SubnetChangeProtection"),
+			},
+
+			{
+				Name:        "encryption_configuration",
+				Description: "A complex type that contains the Amazon Web Services KMS encryption configuration settings for your firewall.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getNetworkFirewallFirewall,
+				Transform:   transform.FromField("Firewall.EncryptionConfiguration"),
+			},
+			{
+				Name:        "firewall_status",
+				Description: "Detailed information about the current status of a Firewall.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getNetworkFirewallFirewall,
+			},
+			{
+				Name:        "subnet_mappings",
+				Description: "The public subnets that Network Firewall is using for the firewall.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getNetworkFirewallFirewall,
+				Transform:   transform.FromField("Firewall.SubnetMappings"),
+			},
+			{
+				Name:        "tags_src",
+				Description: "A list of tags associated with the firewall",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getNetworkFirewallFirewall,
+				Transform:   transform.FromField("Firewall.Tags"),
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getNetworkFirewallFirewall,
+				Transform:   transform.FromField("Firewall.FirewallName"),
+			},
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getNetworkFirewallFirewall,
+				Transform:   transform.FromField("Firewall.Tags").Transform(networkFirewallTurbotTags),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getNetworkFirewallFirewall,
+				Transform:   transform.FromField("Firewall.FirewallArn").Transform(transform.EnsureStringArray),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listNetworkFirewallFirewalls(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+
+	// vpcId := d.EqualsQuals["vpc_id"].GetStringValue()
+
+	// Reduce the basic request limit down if the user has only requested a small number of rows
+	maxLimit := int32(100)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			if limit < 1 {
+				maxLimit = 1
+			} else {
+				maxLimit = limit
+			}
+		}
+	}
+
+	input := &networkfirewall.ListFirewallsInput{
+		MaxResults: &maxLimit,
+	}
+
+	if d.Quals["vpc_id"] != nil {
+		for _, q := range d.Quals["vpc_id"].Quals {
+			value := q.Value.GetStringValue()
+			if q.Operator == "=" {
+				input.VpcIds = append(input.VpcIds, value)
+			}
+		}
+	}
```

Comment:
Could we minimize the above code block as follows?

```
vpcId := d.EqualsQuals["vpc_id"].GetStringValue()
if strings.TrimSpace(vpcId) != "" {
		input.VpcIds = append(input.VpcIds, vpcId)
	}
```

Reply:
This reduces api calls when the query has `in` clause. All in clause ids can be passed to the as the part of request params